examples/advanced/Tutorial3/reconstruction/FairTestDetectorRecoTask.cxx

```diff
@@ -97,7 +97,9 @@ void FairTestDetectorRecoTask::Exec(Option_t* /*opt*/)
         TVector3 dpos(1 / TMath::Sqrt(12), 1 / TMath::Sqrt(12), 1 / TMath::Sqrt(12));
 
         FairTestDetectorHit* hit = new ((*fHitArray)[ipnt]) FairTestDetectorHit(-1, -1, pos, dpos);
-        hit->AddLink(FairLink(-1, FairRootManager::Instance()->GetEntryNr(), FairRootManager::Instance()->GetBranchId("FairTestDetectorDigi"), ipnt));
+        if ( FairRootManager::Instance() != NULL ) {
```

Comment:
Nice discussion. The problem here was that we use the FairTestDetectorRecoTask both in FairRoot (where we have access to input tree via FairRootManager and can set the link) _and_ in FairMQ (where we do not even want to create the FairRootManager and thus cannot set the link). I was quite happy that the FairRootManager::Instance() does not create new FRM, but only returns the pointer to already existing singleton.